src/Microsoft.ApplicationInsights/Extensibility/W3C/W3CUtilities.cs

```diff
@@ -27,6 +29,26 @@ public static string GenerateTraceId()
             return GenerateId(firstHalf, secondHalf, 0, 16);
         }
 
+        /// <summary>
+        /// Constructs a Telemetry ID from given traceid and span id in the format |traceid.spanid.
+        /// This is the format used by Application Insights.        
+        /// </summary>
+        /// <returns>constructed Telemetry ID.</returns>
+        internal static string FormatTelemetryId(string traceId, string spanId)
+        {
+            return string.Concat("|", traceId, ".", spanId, ".");
+        }
+
+        /// <summary>
+        /// Checks if the given string is a valid trace-id as per W3C Specs.
+        /// https://github.com/w3c/distributed-tracing/blob/master/trace_context/HTTP_HEADER_FORMAT.md#trace-id .
+        /// </summary>
+        /// <returns>true if valid w3c trace id, otherwise false.</returns>
+        internal static bool IsCompatibleW3CTraceID(string traceId)
+        {
+            return TraceIdRegex.IsMatch(traceId);
+        }
+
```

Comment:
please update GenerateId methods below to use ActivityTraceId/SpanId.CreateRandom()

Reply:
We need to keep all code so as to fall back if Activity is not available. (standalone use case)

Reply:
generateid methods modified to leverage CreateRandom() if available.